ninshiki_py/detector/tflite.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2021 Ichiro ITS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import cv2
+import json
+import numpy as np
+import os
+from rclpy.node import MsgType
+from tflite_runtime.interpreter import Interpreter
+from tflite_support import metadata
+from ninshiki_interfaces.msg import DetectedObject
+
+
+class TfLite:
+    def __init__(self):
+        self.labels = None
+        self.model = os.path.expanduser('~') + "/tflite_model/efficientdet_lite0.tflite"
+
+        # variable declaration
+        self.interpreter = None
+        self.input_details = None
+        self.output_details = None
+        self.model_width = 0
+        self.model_height = 0
+        self.floating_model = None
+        self.input_mean = 0
+        self.input_std = 0
+
+        self.initiate_tflite()
+
+    def initiate_tflite(self):
+        self.interpreter = Interpreter(model_path=self.model)
+        self.interpreter.allocate_tensors()
+
+        # Get model details
+        self.input_details = self.interpreter.get_input_details()
+        self.output_details = self.interpreter.get_output_details()
+        self.model_height = self.input_details[0]['shape'][1]
+        self.model_width = self.input_details[0]['shape'][2]
+
+        self.floating_model = (self.input_details[0]['dtype'] == np.float32)
+
+        self.load_metadata()
```

Comment:
IDK if it's necessary, but should we add an if guard here? just in case if `len(input_details) < 1` or there's no `'shape'` in `input_details[0]`.

Reply:
yes I think it is necessary